src/Microsoft.VisualStudio.ProjectSystem.Managed/LinqExtensions.cs

```diff
@@ -86,5 +86,17 @@ public static bool Any<T>(this T[] array, Func<T, bool> predicate)
 
             return builder.MoveToImmutable();
         }
+
+        public static TOut[] SelectArray<TIn, TOut>(this TIn[] array, Func<TIn, TOut> selector)
```

Comment:
Is the performance difference with LINQ's `Select` because this version doesn't need to allocate a type from Linq's query engine behind-the-scenes?

Reply:
There savings are:
- No enumerator allocated for `array`
- The `output` array is pre-allocated at the correct size, whereas in the Linq case the `Select` produces an enumerable with no length information such that the chained `ToArray` call will have to buffer items, potentially allocating multiple arrays (doubling the length each time) until the input is exhausted, and then likely allocates another array of the correct length as part of trimming the buffer to size.